x/interchainquery/keeper/msg_server.go

```diff
@@ -30,174 +30,153 @@ func NewMsgServerImpl(keeper Keeper) types.MsgServer {
 
+	// Loop through each module until the callbackId is found in one of the module handlers
 	for _, moduleName := range moduleNames {
-		k.Logger(ctx).Info(fmt.Sprintf("[ICQ Resp] executing callback for queryId (%s), module (%s)", q.Id, moduleName))
 		moduleCallbackHandler := k.callbacks[moduleName]
 
-		if moduleCallbackHandler.HasICQCallback(q.CallbackId) {
-			k.Logger(ctx).Info(fmt.Sprintf("[ICQ Resp] callback (%s) found for module (%s)", q.CallbackId, moduleName))
-			// call the correct callback function
-			err := moduleCallbackHandler.CallICQCallback(ctx, q.CallbackId, msg.Result, q)
-			if err != nil {
-				k.Logger(ctx).Error(fmt.Sprintf("[ICQ Resp] error in ICQ callback, error: %s, msg: %s, result: %v, type: %s, params: %v", err.Error(), msg.QueryId, msg.Result, q.QueryType, q.Request))
-				return err
-			}
-		} else {
-			k.Logger(ctx).Info(fmt.Sprintf("[ICQ Resp] callback not found for module (%s)", moduleName))
+		// Once the callback is found, invoke the function
+		if moduleCallbackHandler.HasICQCallback(query.CallbackId) {
+			return moduleCallbackHandler.CallICQCallback(ctx, query.CallbackId, msg.Result, query)
 		}
 	}
-	return nil
-}
-
-// verify the query has not exceeded its ttl
-func (k Keeper) HasQueryExceededTtl(ctx sdk.Context, msg *types.MsgSubmitQueryResponse, query types.Query) (bool, error) {
-	k.Logger(ctx).Info(fmt.Sprintf("[ICQ Resp] query %s with ttl: %d, resp time: %d.", msg.QueryId, query.Ttl, ctx.BlockHeader().Time.UnixNano()))
-	currBlockTime, err := cast.ToUint64E(ctx.BlockTime().UnixNano())
-	if err != nil {
-		return false, err
-	}
 
-	if query.Ttl < currBlockTime {
-		errMsg := fmt.Sprintf("[ICQ Resp] aborting query callback due to ttl expiry! ttl is %d, time now %d for query of type %s with id %s, on chain %s",
-			query.Ttl, ctx.BlockTime().UnixNano(), query.QueryType, query.ChainId, msg.QueryId)
-		fmt.Println(errMsg)
-		k.Logger(ctx).Error(errMsg)
-		return true, nil
-	}
-	return false, nil
+	// If no callback was found, return an error
+	return types.ErrICQCallbackNotFound
 }
 
+// Handle ICQ query responses by validating the proof, and calling the query's corresponding callback
 func (k msgServer) SubmitQueryResponse(goCtx context.Context, msg *types.MsgSubmitQueryResponse) (*types.MsgSubmitQueryResponseResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
 	// check if the response has an associated query stored on stride
-	q, found := k.GetQuery(ctx, msg.QueryId)
+	query, found := k.GetQuery(ctx, msg.QueryId)
 	if !found {
-		k.Logger(ctx).Info("[ICQ Resp] ignoring non-existent query response (note: duplicate responses are nonexistent)")
+		k.Logger(ctx).Info("ICQ RESPONSE  | Ignoring non-existent query response (note: duplicate responses are nonexistent)")
 		return &types.MsgSubmitQueryResponseResponse{}, nil // technically this is an error, but will cause the entire tx to fail if we have one 'bad' message, so we can just no-op here.
 	}
 
 	defer ctx.EventManager().EmitEvents(sdk.Events{
```

Comment:
Let's make sure we test this thoroughly with the ibc-go relayer, since ibc-go's ICQ relaying assumes on a specific Stride/x/interchainquery event emission schema to detect relayed ICQs. Did you check this against Andrew's schema or test it with the relayer?

Reply:
Was there a change that you thought would be incompatible? The change you've highlighted here is just in the logs - it doesn't modify the events at all

Also we can use the integration tests to confirm the ICQs still work with the relayers

Reply:
Also fwiw, I did test the slash queries still work as expected

Reply:
Discussed this offline. It works!